### DISCUSSION TITLE
WIQL Get all work items since last build

### DISCUSSION THREAD

**Discussion by SimonWoessner** [rfennell/AzurePipelines#1193] [Q&A]
Greetings, 

we are using your extension quite a while and it works like a charme. 

Now we have a different project setup need a list of all workitems since the last build of this specific pipeline. We are able to use WIQL and get all WIs since a date. Could you give us a hint, how to get the date of teh last sucessfull build of this pipeline?

Thanks in advance
Simon 

WIQL:
[System.TeamProject] = ' IBP18599reloaded' AND [Microsoft.VSTS.Common.ClosedDate] > @startOfMonth

**Comment by rfennell** [comment]
You can't get that information via WIQL. However, does not the tasks basic functionality of the task just do the job for you?

By default, the task checks for changes since the last successful build of the current stage in the build. You can override the stage name or the whole build by settings the `overrideBuildReleaseId` setting [See the Usage section of the WIKI](https://github.com/rfennell/AzurePipelines/wiki/GenerateReleaseNotes---Node-based-Cross-Platform-Task#usage-patterns)

**Comment by SimonWoessner** [comment]
Hi, wow. that was a fast answer!

As far as I understood, it only shows Workitems linked to commits, We are not linking the WorkItems to commits in our multi pipeline setup. 

I know about overrideBuildReleaseId, but that is something else. 


Is there a setting, that returns all Workitems since the last build? 

Otherwise we will still use the WIQL and postprocess the file, deleting all "older" WorkItems.

Best
Simon

**Reply by rfennell** [reply]
No sorry, there is no current functionality to help. The task is very much designed around the assumption work items are associated with commits/checkins and hence builds.

**Comment by SimonWoessner** [comment]
All good. 
We can postprocess the data without problems. 
Nevertheless a great and usefull extension, which we definitely will keep using

**Reply by SimonWoessner** [reply]
Apologies, but I have one more question.

Is it possible, that we use a pipeline variable in the WIQL statement. With that we most probably could solve everything?

Best and thank you!

**Reply by rfennell** [reply]
Yes, can't see why not. You pass in the WIQL as a parameter and this string is evaluated at runtime so you can use predefined variables or your own customer ones e.g.

```
- task: XplatGenerateReleaseNotes@3
  inputs: 
     {all the other params}
    wiqlWhereClause: "[System.TeamProject] = '$(System.TeamProject)' and [System.WorkItemType] = '$(MY-OWN-VARIABLE)'"  
   ```
